pkg/syncer/spec/spec_controller.go

```diff
@@ -96,7 +99,56 @@ func NewSpecSyncer(gvrs []schema.GroupVersionResource, upstreamClusterName logic
 				c.AddToQueue(gvr, obj)
 			},
 		})
-		klog.InfoS("Set up informer", "clusterName", upstreamClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
+		klog.InfoS("Set up upstream informer", "clusterName", upstreamClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
+
+		downstreamInformers.ForResource(gvr).Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+			DeleteFunc: func(obj interface{}) {
+				unstrob := obj.(*unstructured.Unstructured)
+				klog.InfoS("downstream delete event", "gvr", gvr, "namespace", unstrob.GetNamespace(), "name", unstrob.GetName())
+				//if there is no label then it is not owned by kcp
+				if unstrob.GetLabels() == nil {
```

Comment:
This is enforced by the list options going into the informer, so why do we need this check? Also, check the key-value, not for the existence of labels in general.